models/DataObject/ClassDefinition/LinkGeneratorInterface.php

```diff
@@ -20,6 +20,12 @@
 interface LinkGeneratorInterface
 {
     /**
+     *
+     * If you want to support mockups or arbitrary objects you can change the typehint to
+     * public function generate(object $object, array $params = []): string
```

Comment:
Do we really need this comment? That's basic PHP contravariance behavior, so you could basically add this to every method...